### PR TITLE
ARROW-15885 [Ruby] Add support for #values of DayTime Interval Type

### PR DESCRIPTION
**Pull request by okadakk**
Implement #values method of day time interval type to red arrow.

### CODE REVIEW

File: ruby/red-arrow/ext/arrow/converters.hpp [hunk 1]
```diff
@@ -207,6 +207,21 @@ namespace red_arrow {
       return INT2NUM(array.Value(i));
     }
 
+    inline VALUE convert(const arrow::DayTimeIntervalArray& array,
+                         const int64_t i) {
+
```
Comment (kou) [inline ruby/red-arrow/ext/arrow/converters.hpp:212]:
```suggestion
```

File: ruby/red-arrow/ext/arrow/converters.hpp [hunk 2]
```diff
@@ -207,6 +207,21 @@ namespace red_arrow {
...
+      auto value = rb_hash_new();
+      auto arrow_value = array.Value(i);
+      rb_hash_aset(value,
+                   red_arrow::symbols::day,
+                   INT2NUM(arrow_value.days));
+      rb_hash_aset(value,
+                   red_arrow::symbols::millisecond,
+                   INT2NUM(arrow_value.milliseconds));
+
```
Comment (kou) [inline ruby/red-arrow/ext/arrow/converters.hpp:221]:
```suggestion
```

File: ruby/red-arrow/test/values/test-basic-arrays.rb [hunk 3]
```diff
@@ -286,6 +286,21 @@ def test_month_interval
...
+      {day: 1, millisecond: 100},
+      nil,
+      {day: 2, millisecond: 300},
+    ]
+    inputs = [
+      Arrow::DayMillisecond.new(1, 100),
+      nil,
+      Arrow::DayMillisecond.new(2, 300),
+    ]
+    target = build(Arrow::DayTimeIntervalArray.new(inputs))
```
Comment (kou) [inline ruby/red-arrow/test/values/test-basic-arrays.rb:301]: Ah, we should add support for `Arrow::DayTimeIntervalArray.new([{day: 1, millisecond: 100}, nil, {...}])`.
Could you open an issue for this improvement?

**Review by kou — APPROVED** [review]
+1

File: ruby/red-arrow/test/values/test-basic-arrays.rb (hunk 3 above)
Comment (okadakk) [reply ruby/red-arrow/test/values/test-basic-arrays.rb:301]: OK!
I opened it!
https://issues.apache.org/jira/browse/ARROW-15918 

And I challenge to support this!!

**Pull request closed without merging**